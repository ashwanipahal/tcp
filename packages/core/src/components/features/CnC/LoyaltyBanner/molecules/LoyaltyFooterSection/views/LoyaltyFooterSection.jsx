import React from 'react';
import { PropTypes } from 'prop-types';
import ClickTracker from '@tcp/web/src/components/common/atoms/ClickTracker';
import BagPageUtils from '@tcp/core/src/components/features/CnC/BagPage/views/Bagpage.utils';
import withStyles from '../../../../../../common/hoc/withStyles';
import Styles from '../styles/LoyaltyFooterSection.style';
import { BodyCopy, Anchor } from '../../../../../../common/atoms';

const openModalApplyNowModal = (openApplyNowModal, step = 1) => {
  if (openApplyNowModal) {
    if (step === 1) {
      return openApplyNowModal({ isModalOpen: true, status: null });
    }
    return openApplyNowModal({ isModalOpen: false, isPLCCModalOpen: true, status: null });
  }
  return null;
};

const renderApplyNowLink = (text, closeAddedToBagModal, openApplyNowModal, cartOrderItems) => {
  const pageData = 'shopping bag';
  const productsData = BagPageUtils.formatBagProductsData(cartOrderItems);
  return (
    <Anchor
      fontSizeVariation="medium"
      anchorVariation="primary"
      noLink
      handleLinkClick={e => {
        e.preventDefault();
        openModalApplyNowModal(openApplyNowModal, 2);
        closeAddedToBagModal();
      }}
      underline
    >
      <ClickTracker
        clickData={{
          pageType: pageData,
          pageSection: pageData,
          pageSubSection: pageData,
          products: productsData,
          pageName: pageData,
          customEvents: ['event116'],
          eventName: 'loyaltyclick',
        }}
      >
        {text}
      </ClickTracker>
    </Anchor>
  );
};

const renderLearnMoreLink = (text, closeAddedToBagModal, openApplyNowModal, cartOrderItems) => {
  const productsData = BagPageUtils.formatBagProductsData(cartOrderItems);
  return (
    <Anchor
      fontSizeVariation="medium"
      anchorVariation="primary"
      noLink
      underline
      handleLinkClick={e => {
        e.preventDefault();
        openModalApplyNowModal(openApplyNowModal);
        closeAddedToBagModal();
      }}
    >
      <ClickTracker
        clickData={{
          pageName: 'shopping bag',
          eventName: 'loyaltyclick',
          customEvents: ['event117'],
          products: productsData,
        }}
      >
        {text}
      </ClickTracker>
    </Anchor>
  );
};

const onLinkClick = ({ e, componentId, closeAddedToBagModal, openOverlay }) => {
  e.preventDefault();
  closeAddedToBagModal();
  openOverlay({
    component: componentId,
    variation: 'primary',
  });
};

const renderCreateAccountLink = (text, closeAddedToBagModal, openOverlay) => {
  return (
    <Anchor
      fontSizeVariation="medium"
      anchorVariation="primary"
      noLink
      underline
      handleLinkClick={e => {
        e.preventDefault();
        onLinkClick({ e, componentId: 'createAccount', closeAddedToBagModal, openOverlay });
      }}
    >
      {text}
    </Anchor>
  );
};

const renderLoginLink = (text, closeAddedToBagModal, openOverlay) => {
  return (
    <Anchor
      fontSizeVariation="medium"
      anchorVariation="primary"
      noLink
      underline
      handleLinkClick={e => {
        e.preventDefault();
        onLinkClick({ e, componentId: 'login', closeAddedToBagModal, openOverlay });
      }}
    >
      {text}
    </Anchor>
  );
};

const getLinkWithName = (props, action, text) => {
  let returnLink;
  const { closeAddedToBagModal, openOverlay, openApplyNowModal, cartOrderItems } = props;
  switch (action) {
    case 'ApplyNowAction':
      returnLink = renderApplyNowLink(
        text,
        closeAddedToBagModal,
        openApplyNowModal,
        cartOrderItems
      );
      break;
    case 'LearnMoreAction':
      returnLink = renderLearnMoreLink(
        text,
        closeAddedToBagModal,
        openApplyNowModal,
        cartOrderItems
      );
      break;
    case 'CreateAccountAction':
      returnLink = renderCreateAccountLink(text, closeAddedToBagModal, openOverlay);
      break;
    case 'loginAction':
      returnLink = renderLoginLink(text, closeAddedToBagModal, openOverlay);
      break;
    default:
      break;
  }
  return returnLink;
};

const LoyaltyFooterSection = props => {
  const { className, footerLabels } = props;
  return (
    <div className={`${className} footer-wrapper`}>
      <div className="links-wrapper">
        {footerLabels.link1Prefix && (
          <BodyCopy
            className="symbolWrapper"
            color="text.primary"
            component="span"
            fontFamily="secondary"
            fontWeight="regular"
            fontSize="fs9"
          >
            {footerLabels.link1Prefix}
          </BodyCopy>
        )}
        {footerLabels.link1Action &&
          getLinkWithName(props, footerLabels.link1Action, footerLabels.link1Text)}
        {footerLabels.link1Action && footerLabels.link2Action && <span className="space-between" />}
        {footerLabels.link2Prefix && (
          <BodyCopy
            className="symbolWrapper"
            color="text.primary"
            component="span"
            fontFamily="secondary"
            fontWeight="regular"
            fontSize="fs9"
          >
            {footerLabels.link2Prefix}
          </BodyCopy>
        )}
        {footerLabels.link2Action &&
          getLinkWithName(props, footerLabels.link2Action, footerLabels.link2Text)}
      </div>
    </div>
  );
};

LoyaltyFooterSection.propTypes = {
  footerLabels: PropTypes.shape.isRequired,
  className: PropTypes.string,
};

LoyaltyFooterSection.defaultProps = {
  className: '',
};

export default withStyles(LoyaltyFooterSection, Styles);
export { LoyaltyFooterSection as LoyaltyFooterSectionVanilla };
