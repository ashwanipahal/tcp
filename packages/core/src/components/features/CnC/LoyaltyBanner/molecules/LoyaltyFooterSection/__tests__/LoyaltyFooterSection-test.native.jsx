import React from 'react';
import { shallow } from 'enzyme';
import { LoyaltyFooterSectionVanilla } from '../views/LoyaltyFooterSection.native';

describe('LoyaltyFooterSection', () => {
  let props;
  beforeEach(() => {
    props = {
      className: '',
      footerLabels: {},
      openApplyNowModal: jest.fn(),
      closeAddedToBagModal: jest.fn(),
      navigation: {
        navigate: jest.fn(),
      },
    };
  });
  let component;
  it('should render correctly', () => {
    component = shallow(<LoyaltyFooterSectionVanilla {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('isProductDetailView should render correctly', () => {
    props.footerLabels.link1Text = 'abc';
    props.footerLabels.link1Action = 'abc';
    props.footerLabels.link1Prefix = 'abc';
    props.footerLabels.link2Text = 'abc';
    props.footerLabels.link2Action = 'abc';
    props.footerLabels.link2Prefix = 'abc';
    component = shallow(<LoyaltyFooterSectionVanilla {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('isProductDetailView && plcc && isGuest should render correctly', () => {
    props.footerLabels.link1Text = 'ApplyNowAction';
    props.footerLabels.link1Action = 'ApplyNowAction';
    props.footerLabels.link1Prefix = 'abc';
    props.footerLabels.link2Text = 'LearnMoreAction';
    props.footerLabels.link2Action = 'LearnMoreAction';
    props.footerLabels.link2Prefix = 'abc';
    component = shallow(<LoyaltyFooterSectionVanilla {...props} />);
    component
      .find('Styled(Anchor)')
      .at(0)
      .simulate('press');
    component
      .find('Styled(Anchor)')
      .at(1)
      .simulate('press');
    expect(component).toMatchSnapshot();
  });

  it('isProductDetailView && isGuest should render correctly', () => {
    props.footerLabels.link1Text = 'CreateAccountAction';
    props.footerLabels.link1Action = 'CreateAccountAction';
    props.footerLabels.link1Prefix = 'abc';
    props.footerLabels.link2Text = 'loginAction';
    props.footerLabels.link2Action = 'loginAction';
    props.footerLabels.link2Prefix = 'abc';
    component = shallow(<LoyaltyFooterSectionVanilla {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('should render correctly if show modal', () => {
    component = shallow(<LoyaltyFooterSectionVanilla {...props} />);
    component.setState({ showModal: true });
    expect(component).toMatchSnapshot();
  });
  it('should call toggleModal with getComponentId ', () => {
    component = shallow(<LoyaltyFooterSectionVanilla {...props} />);
    component.setState({ showModal: false });
    component
      .instance()
      .toggleModal({ getComponentId: { login: 'login', createAccount: 'create' } });
    expect(component.state('showModal')).toBe(true);
    expect(component.state('getComponentId')).toMatchObject({
      login: 'login',
      createAccount: 'create',
    });
  });
  it('should call toggleModal without getComponentId ', () => {
    component = shallow(<LoyaltyFooterSectionVanilla {...props} />);
    component.setState({ showModal: false });
    component.instance().toggleModal({});
    expect(component.state('showModal')).toBe(true);
    expect(component.state('getComponentId')).toBe('');
  });
});
