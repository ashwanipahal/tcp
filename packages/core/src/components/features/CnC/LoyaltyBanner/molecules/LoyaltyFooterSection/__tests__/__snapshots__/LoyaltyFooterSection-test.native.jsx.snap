// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoyaltyFooterSection isProductDetailView && isGuest should render correctly 1`] = `
<View
  className=" footer-wrapper"
>
  <Styled(View)>
    <Styled(BodyCopy)
      accessibilityText=""
      color="text.primary"
      dataLocator=""
      fontSize="fs9"
      fontWeight="regular"
      margin={null}
      mobileFontFamily="secondary"
      numberOfLines={null}
      text="abc"
      textDecoration={null}
    />
    <Styled(Anchor)
      accessibilityLabel=""
      anchorVariation="primary"
      className="createAccount"
      colorName={null}
      customStyle={Object {}}
      fontSizeVariation="medium"
      justifyContent=""
      locator=""
      margins={null}
      onPress={[Function]}
      openWebView={false}
      text="CreateAccountAction"
      underline={true}
      visible={false}
    />
    <Styled(View) />
    <Styled(BodyCopy)
      accessibilityText=""
      color="text.primary"
      dataLocator=""
      fontSize="fs9"
      fontWeight="regular"
      margin={null}
      mobileFontFamily="secondary"
      numberOfLines={null}
      text="abc"
      textDecoration={null}
    />
    <Styled(Anchor)
      accessibilityLabel=""
      anchorVariation="primary"
      className="logIn"
      colorName={null}
      customStyle={Object {}}
      fontSizeVariation="medium"
      justifyContent=""
      locator=""
      margins={null}
      onPress={[Function]}
      openWebView={false}
      text="loginAction"
      underline={true}
      visible={false}
    />
  </Styled(View)>
</View>
`;

exports[`LoyaltyFooterSection isProductDetailView && plcc && isGuest should render correctly 1`] = `
<View
  className=" footer-wrapper"
>
  <Styled(View)>
    <Styled(BodyCopy)
      accessibilityText=""
      color="text.primary"
      dataLocator=""
      fontSize="fs9"
      fontWeight="regular"
      margin={null}
      mobileFontFamily="secondary"
      numberOfLines={null}
      text="abc"
      textDecoration={null}
    />
    <Styled(Anchor)
      accessibilityLabel=""
      anchorVariation="primary"
      className="applyNow"
      colorName={null}
      customStyle={Object {}}
      fontSizeVariation="medium"
      justifyContent=""
      locator=""
      margins={null}
      onPress={[Function]}
      openWebView={false}
      text="ApplyNowAction"
      underline={true}
      visible={false}
    />
    <Styled(View) />
    <Styled(BodyCopy)
      accessibilityText=""
      color="text.primary"
      dataLocator=""
      fontSize="fs9"
      fontWeight="regular"
      margin={null}
      mobileFontFamily="secondary"
      numberOfLines={null}
      text="abc"
      textDecoration={null}
    />
    <Styled(Anchor)
      accessibilityLabel=""
      anchorVariation="primary"
      className="learnMore"
      colorName={null}
      customStyle={Object {}}
      fontSizeVariation="medium"
      justifyContent=""
      locator=""
      margins={null}
      onPress={[Function]}
      openWebView={false}
      text="LearnMoreAction"
      underline={true}
      visible={false}
    />
  </Styled(View)>
</View>
`;

exports[`LoyaltyFooterSection isProductDetailView should render correctly 1`] = `
<View
  className=" footer-wrapper"
>
  <Styled(View)>
    <Styled(BodyCopy)
      accessibilityText=""
      color="text.primary"
      dataLocator=""
      fontSize="fs9"
      fontWeight="regular"
      margin={null}
      mobileFontFamily="secondary"
      numberOfLines={null}
      text="abc"
      textDecoration={null}
    />
    <Styled(View) />
    <Styled(BodyCopy)
      accessibilityText=""
      color="text.primary"
      dataLocator=""
      fontSize="fs9"
      fontWeight="regular"
      margin={null}
      mobileFontFamily="secondary"
      numberOfLines={null}
      text="abc"
      textDecoration={null}
    />
  </Styled(View)>
</View>
`;

exports[`LoyaltyFooterSection should render correctly 1`] = `
<View
  className=" footer-wrapper"
>
  <Styled(View) />
</View>
`;

exports[`LoyaltyFooterSection should render correctly if show modal 1`] = `
<View
  className=" footer-wrapper"
>
  <Styled(View)>
    <ModalNative
      fontSize="fs16"
      heading=" "
      headingFontFamily="secondary"
      horizontalBar={true}
      inheritedStyles=""
      isOpen={true}
      isOverlay={false}
      onRequestClose={[Function]}
    >
      <View />
    </ModalNative>
  </Styled(View)>
</View>
`;
